### PR TITLE
KV / integrity checker: conflate `Disputed`, `InvalidLedgerTime` and `Inconsistent` rejection reasons [KVL-1059]

### PR DESCRIPTION
**Pull request by fabiotudone-da**
# Description:

Rejection in transaction validation has been re-designed in #10066 and the resulting KV write set is (intentionally and acceptedly) incompatible with the previous one, requiring a more relaxed checking.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by nicu-da — APPROVED** [review]
LGTM!

Just wondering if there's a way to add tests that cover this scenario so we catch it beforehand.

**Comment by fabiotudone-da** [comment]
> LGTM!
> 
> Just wondering if there's a way to add tests that cover this scenario so we catch it beforehand.

Yes, we should really check against historical exports. We'll do that at some point.

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala [hunk 1]
```diff
@@ -72,7 +73,24 @@ object RejectionReasonDescriptionNormalizer extends UpdateNormalizer {
   }
 }
 
-/** Rejection reason strings should always be ignored as they can change arbitrarily. */
+/** Rejection reasons `Disputed`, `InvalidLedgerTime` and `Inconsistent` are being conflated
+  *  as their usage has been re-designed in SDK 1.16.0-snapshot.20210727.7476.0.b5e9d861 [KVL-1015].
+  */
+object RejectionReasonNormalizer extends UpdateNormalizer {
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala:79]: Can we come up with a more specific name? Perhaps `RejectionsForConsistencyNormalizer`?

**Review by SamirTalwar-DA — APPROVED** [review]

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala (hunk 1 above)
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala:79]: What about `CommandRejectionRedesignNormalizer`?

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala:79]: (I know it's a history-dependent name but updates normalizers are inherently historical byproducts)

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala:79]: Sure, works for me.

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/UpdateNormalizer.scala]: 399012a

**Review by hubert-da — APPROVED** [review]

**Pull request merged**